File: Luca/app/src/main/java/de/culture4life/luca/network/pojo/LocationResponseData.java
package de.culture4life.luca.network.pojo;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class LocationResponseData implements Serializable {

    public enum EntryPolicy {
        @SerializedName("2G") POLICY_2G,
        @SerializedName("3G") POLICY_3G
    }

    @SerializedName("locationId")
    private String locationId;

    @SerializedName("locationName")
    private String areaName;

    @SerializedName("groupName")
    private String groupName;

    @SerializedName("lat")
    private double latitude;

    @SerializedName("lng")
    private double longitude;

    @SerializedName("radius")
    private long radius;

    @SerializedName("isPrivate")
    private boolean isPrivate;

    @SerializedName("isContactDataMandatory")
    private boolean isContactDataMandatory = true;

    @SerializedName("entryPolicy")
    private EntryPolicy entryPolicy = null;

    @SerializedName("averageCheckinTime")
    private long averageCheckInDuration;

    public String getLocationId() {
        return locationId;
    }

    public void setLocationId(String locationId) {
        this.locationId = locationId;
    }

    public String getAreaName() {
        return areaName;
    }

    public void setAreaName(String areaName) {
        this.areaName = areaName;
    }

    public String getGroupName() {
        return groupName;
    }

    public void setGroupName(String groupName) {
        this.groupName = groupName;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public long getRadius() {
        return radius;
    }

    public void setRadius(long radius) {
        this.radius = radius;
    }

    public boolean isPrivate() {
        return isPrivate;
    }

    public void setPrivate(boolean isPrivate) {
        this.isPrivate = isPrivate;
    }

    public boolean isContactDataMandatory() {
        return isContactDataMandatory;
    }

    public void setIsContactDataMandatory(boolean isMandatory) {
        this.isContactDataMandatory = isMandatory;
    }

    public EntryPolicy getEntryPolicy() {
        return entryPolicy;
    }

    public void setEntryPolicy(EntryPolicy entryPolicy) {
        this.entryPolicy = entryPolicy;
    }

    public long getAverageCheckInDuration() {
        return averageCheckInDuration;
    }

    public void setAverageCheckInDuration(long averageCheckInDuration) {
        this.averageCheckInDuration = averageCheckInDuration;
    }

    @Override
    public String toString() {
        return "LocationResponseData{" +
                "locationId='" + locationId + '\'' +
                ", areaName='" + areaName + '\'' +
                ", groupName='" + groupName + '\'' +
                ", latitude=" + latitude +
                ", longitude=" + longitude +
                ", radius=" + radius +
                ", isPrivate=" + isPrivate +
                ", isContactDataMandatory=" + isContactDataMandatory +
                ", entryPolicy=" + entryPolicy +
                ", averageCheckInDuration=" + averageCheckInDuration +
                '}';
    }
}

File: Luca/app/src/main/java/de/culture4life/luca/network/pojo/LocationResponseData.kt
package de.culture4life.luca.network.pojo

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class LocationResponseData(

    @SerializedName("locationId")
    val locationId: String? = null,

    @SerializedName("locationName")
    val areaName: String? = null,

    @SerializedName("groupName")
    val groupName: String? = null,

    @SerializedName("lat")
    val latitude: Double = 0.0,

    @SerializedName("lng")
    val longitude: Double = 0.0,

    @SerializedName("radius")
    val radius: Long = 0,

    @SerializedName("isPrivate")
    val isPrivate: Boolean = false,

    @SerializedName("isContactDataMandatory")
    val isContactDataMandatory: Boolean = true,

    @SerializedName("entryPolicy")
    val entryPolicy: EntryPolicy? = null,

    @SerializedName("averageCheckinTime")
    val averageCheckInDuration: Long = 0
) : Serializable {
    enum class EntryPolicy {
        POLICY_2G, POLICY_3G
    }
}
